src/opm/input/eclipse/EclipseState/SimulationConfig/BCConfig.cpp

```diff
@@ -64,18 +64,41 @@ BCComponent component(const std::string& s) {
 }
 }
 
-BCConfig::BCFace::BCFace(const DeckRecord& record) :
-    i1(record.getItem("I1").get<int>(0) - 1),
-    i2(record.getItem("I2").get<int>(0) - 1),
-    j1(record.getItem("J1").get<int>(0) - 1),
-    j2(record.getItem("J2").get<int>(0) - 1),
-    k1(record.getItem("K1").get<int>(0) - 1),
-    k2(record.getItem("K2").get<int>(0) - 1),
+BCConfig::BCFace::BCFace(const DeckRecord& record, const GridDims& grid) :
+    i1(0),
+    i2(grid.getNX() - 1),
+    j1(0),
+    j2(grid.getNY() - 1),
+    k1(0),
+    k2(grid.getNZ() - 1),
     bctype(fromstring::bctype(record.getItem("TYPE").get<std::string>(0))),
     dir(FaceDir::FromString(record.getItem("DIRECTION").get<std::string>(0))),
     component(fromstring::component(record.getItem("COMPONENT").get<std::string>(0))),
     rate(record.getItem("RATE").getSIDouble(0))
 {
+    const auto& itemI1 = record.getItem("I1");
+    if(!itemI1.defaultApplied(0))
+        i1 = itemI1.get<int>(0) - 1;
+
+    const auto& itemI2 = record.getItem("I2");
+    if(!itemI2.defaultApplied(0))
+        i2 = itemI2.get<int>(0) - 1;
+
+    const auto& itemJ1 = record.getItem("J1");
+    if(!itemJ1.defaultApplied(0))
+        j1 = itemJ1.get<int>(0) - 1;
+
+    const auto& itemJ2 = record.getItem("J2");
+    if(!itemJ2.defaultApplied(0))
+        j2 = itemJ2.get<int>(0) - 1;
+
+    const auto& itemK1 = record.getItem("K1");
+    if(!itemK1.defaultApplied(0))
+        k1 = itemK1.get<int>(0) - 1;
+
+    const auto& itemK2 = record.getItem("K2");
+    if(!itemK2.defaultApplied(0))
+        k2 = itemK2.get<int>(0) - 1;
```

Comment:
Would you mind switching to compiled items here, please?  Something along the lines of
```C++
using BC = ParserKeyword::BC;
const auto& itemI1 = record.getItem<BC::I1>();
if (! itemI1.defaultApplied(0)) {
    this->i1 = itemI1.get<int>(0) - 1;
}
```
and so on?  That way we get a **little** more safety if the item names happen to change in the future.

We might even consider going further to use `if-init`
```C++
using BC = ParserKeywords::BC;
if (const auto& I1 = record.getItem<BC::I1>(); ! I1.defaultApplied(0)) {
    this->i1 = I1.get<int>(0) - 1;
}
```
and so on.

Reply:
Thanks. if-init was a good idea, it avoid the danger of copy-paste error since I1 is only available within the brackets.